dashboard/app/controllers/courses_controller.rb

```diff
@@ -20,4 +30,20 @@ def show
 
     render 'show', locals: {course: course}
   end
+
+  def new
+  end
+
+  def create
+    course = Course.new(name: params.require(:course).require(:name))
+    if course.save
+      redirect_to action: :edit, course_name: course.name
+    else
+      render 'new', locals: {course: course}
+    end
+  end
```

Comment:
I'm not sure how interesting this method is to test. I might add a quick and dirty test in a future PR.